src/ddsrt/include/dds/ddsrt/ifaddrs.h

```diff
@@ -24,41 +24,47 @@ extern "C" {
 
 /** @brief The interface type */
 enum ddsrt_iftype {
-  DDSRT_IFTYPE_UNKNOWN,
-  DDSRT_IFTYPE_WIRED,
-  DDSRT_IFTYPE_WIFI
+  DDSRT_IFTYPE_UNKNOWN, ///< An unknown interface
+  DDSRT_IFTYPE_WIRED, ///< A wired interface (e.g. ethernet)
+  DDSRT_IFTYPE_WIFI ///< A wifi interface
 };
 
 /** @brief Linked list of interface addresses */
 struct ddsrt_ifaddrs {
-  struct ddsrt_ifaddrs *next;
-  char *name;
-  uint32_t index;
-  uint32_t flags;
-  enum ddsrt_iftype type;
-  struct sockaddr *addr;
-  struct sockaddr *netmask;
-  struct sockaddr *broadaddr;
+  struct ddsrt_ifaddrs *next; ///< next pointer of the linked list
```

Comment:
Wouldn't that be
```suggestion
  struct ddsrt_ifaddrs *next; ///< pointer to the next interface in the returned list of interfaces
```
I am aware it is pretty much the same, but it doesn't hurt to make an effort to make it easier for users ...

(And yes, I am aware that I often fall short of that in writing documentation, but let's try to raise the bar.)

Reply:
I'm not sure what this adds. Surely I can assume the reader knows what a linked list is? Then the next pointer needs no further explanation. It is just a pointer to another struct of the same type.